### PR TITLE
Invoke OnPeerConnectionClosed when ICE connection state changes to closed.

### PR DESCRIPTION
**Pull request by jianjunz**

### CODE REVIEW

**Review by taste1981 — DISMISSED** [review]

File: talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.h [hunk 1]
```diff
@@ -22,6 +22,7 @@ class P2PPeerConnectionChannelObserverObjcImpl
   void OnStreamAdded(
       std::shared_ptr<owt::base::RemoteStream> stream) override;
   void OnStopped(const std::string& remote_id) override;
+  void OnPeerConnectionClosed(const std::string& remote_id) override;
```
Comment (dvrogozh) [inline talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.h:25]: Other places where we use this method are protected with `#ifdef OWT_CG_SERVER`, like here:
https://github.com/open-webrtc-toolkit/owt-client-native/blob/adad03d829d7810b0093fb17f24afe43744d1fab/talk/owt/sdk/p2p/p2pclient.cc#L456
Should this protection be added here as well?

File: talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.mm [hunk 2]
```diff
@@ -37,6 +37,12 @@
   [_observer onStoppedFrom:[NSString stringWithUTF8String:remote_id.c_str()]];
 }
 
+void P2PPeerConnectionChannelObserverObjcImpl::OnPeerConnectionClosed(
```
Comment (dvrogozh) [inline talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.mm:40]: same as above

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.h (hunk 1 above)
Comment (dvrogozh) [reply talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.h:25]: https://github.com/open-webrtc-toolkit/owt-client-native/blob/adad03d829d7810b0093fb17f24afe43744d1fab/talk/owt/sdk/include/cpp/owt/p2p/p2pclient.h#L67

Quoting @taste1981 offline reply: "the code line you're listing is API for P2PClientOsbserver, not the same API jianjun is handling. There're multiple API with the same name on different classes."

Comment (jianjunz) [reply talk/owt/sdk/p2p/objc/P2PPeerConnectionChannelObserverObjcImpl.h:25]: There are two places named `OnPeerConnectionClosed`, one is public (in `P2PClientObserver`). It's protected by `OWT_CG_SERVER`. Another one is not public, defined in `P2PPeerConnectionChannelObserver`. `P2PPeerConnectionChannelObserverObjcImpl` implements the latter one.

**Review by taste1981 — APPROVED** [review]
Still +1 to the change.

**Pull request merged**